_posts/scyrill/11-23-2021-scyrill-project-update.md

```diff
@@ -0,0 +1,158 @@
+```python 
+#look for the lines that have discharge diagnosis 
+for line in pt_notes: 
+  #figure out why this isn't printing 'final diagnosis'
+  If line.startswith(‘Discharge Diagnosis’):
+  print (line)
+```
+It would print all the lines that had ‘Discharge Diagnosis’, which I thought was good, but then I realized that in the file, it also had ‘Final Diagnosis’, ‘Final Diagnoses’, and ‘Discharge Diagnoses’. I learned that the reason why the other phrases weren’t being included was because some of them were in in all capital letters. To solve this, I just included them in a long line that tried to grab all the instances of the word. 
+
+```python 
+#look for the lines that have discharge diagnosis 
+for line in pt_notes: 
+  #figure out why this isn't printing 'final diagnosis'
+  If line.startswith(‘Discharge Diagnosis’) or line.startswith(‘Discharge Diagnoses’) or line.startswith(Final Diagnosis’) or line.startswith(‘Final Diagnoses’):
+  print (line)
+```
+I didn’t like the list of ors in the line because it was long, so I thought the best way to capture it in just a few words would be by making all the phrases uppercase and then catching the lines that way. 
+
+```python 
+#look for the lines that have discharge diagnosis 
+for line in pt_notes: 
+  #figure out why this isn't printing 'final diagnosis'
+  If line.startswith.upper(‘Discharge Diagnosis’) or line.startswith.upper(‘Discharge Diagnoses’) or line.startswith.upper(Final Diagnosis’) or line.startswith.upper(‘Final Diagnoses’):
+  print (line)
+```
+Not only was this longer, it didn’t work. Therefore, I turned to ```regex``` because I knew that it would at least be able to get it. Since it wasn’t catching the other ‘(blank) Diagnos(i/e)s’, I tried using regex to get all the instances where it begins with ______ Diagnos(i/e)s to get all the instances. To get a regex to do this, I searched the file for lines that started with any character, have a whitespace, accepted ‘diagnos’, followed by any character being acceptable to account for i/e, and end with ‘s’. I turned back to chapter 12 on Regular Expressions, and figured that the way to write the code would be using ```re.findall()```. I had used it in the past for homework and exercises, so I figured that it could be used here too.  My code looked something like this: 
+```python 
+for line in pt_notes: 
+  #figure out why this isn't printing 'final diagnosis'
+  re.findall(‘.+\s Diagnos.s’):
```

Comment:
The fact that this line is highlighted in wine red makes me think something is amiss here in your post and possibly in your Trinket code. These angled single quotes `‘` are separate characters from straight single quotes `'`. Same is true for angled `“` vs. straight `"` double quotes. Could you replace them and see how things look/run?

Reply:
Sure, thanks for checking this for me, I noticed this, but wasn't sure if it would cause a problem or not.